src/workerd/api/streams/queue.c++

```diff
@@ -251,16 +251,21 @@ void maybeInvalidateByobRequest(kj::Maybe<ByteQueue::ByobRequest&>& req) {
 }  // namespace
 
 ByteQueue::ReadRequest::ReadRequest(
+    jsg::Lock& js,
     jsg::Promise<ReadResult>::Resolver resolver,
     ByteQueue::ReadRequest::PullInto pullInto)
     : resolver(kj::mv(resolver)),
-      pullInto(kj::mv(pullInto)) {}
+      pullInto(kj::mv(pullInto)) {
+  // The store needs to be accounted for as external memory in the isolate.
+  js.v8Isolate->AdjustAmountOfExternalAllocatedMemory(this->pullInto.store.size());
+}
```

Comment:
For reviewers: There is arguably an easier approach we can take here. The only reason we need to call `AdjustAmountOfExternalAllocatedMemory` here and when the entry is popped from the queue is because at this level we are currently only storing and handling the detached `v8::BackingStore` instances. If we changed this queue to store `v8::ArrayBuffer` instances then the memory accounting would be automatically tracked. That comes at the cost of minor additional allocations on the v8 heap but that may be ok? I'm not in a rush to land this so let's consider the options.

Reply:
The more I think about it... the more I think taking this alternative approach is the better path. I'm going to move this back into draft PR status until I can work that up.

Reply:
Depends how confident you are you can test all the edge cases vs. the potential overhead of the `v8::ArrayBuffer`s I guess. If we think the overhead is relatively low (hopefully is for typical read sizes?) then I'd definitely agree that it makes sense to opt for something which eliminates our need to manually account for the memory usage.

Are you proposing to also use `v8::ArrayBuffer` to account for the period of untracked memory in `ReadableStreamInternalController::read` too? If so, sounds good. If not, I was going to propose at least wrapping the symmetric calls to `AdjustAmountOfExternalAllocatedMemory` in some kind of RAII helper to avoid the possibility of incrementing but not decrementing or vice versa.

Reply:
I plan to split the effort in two parts. First, focusing only on what is stored in the queues, then look separately at the buffers as they are in motion via the various promises. It likely makes sense to use the `v8::ArrayBuffer` wrapper in every case as it'll just make things easier, but I need to verify. The `v8::ArrayBuffer` itself is only going to be a handful of additional bytes, they aren't that big and will be tracked against the heap limit so it's safe.